File: benchmarks/matmul/matmul.py
def main():
    n = 100
    a = [1] * (n * n)
    b = [2] * (n * n)
    c = [0] * (n * n)
    for _ in range(30):
        for i in range(n):
            for j in range(n):
                s = 0
                for k in range(n):
                    s += a[i * n + k] * b[k * n + j]
                c[i * n + j] = s
    print("Matmul done:", c[n * n - 1])

main()

File: benchmarks/matmul/matmul.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int main() {
    int n = 100;
    int64_t *a = malloc(n * n * sizeof(int64_t));
    int64_t *b = malloc(n * n * sizeof(int64_t));
    int64_t *c = calloc(n * n, sizeof(int64_t));
    for (int i = 0; i < n * n; i++) { a[i] = 1; b[i] = 2; }
    for (int iter = 0; iter < 30; iter++) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                int64_t sum = 0;
                for (int k = 0; k < n; k++) {
                    sum += a[i * n + k] * b[k * n + j];
                }
                c[i * n + j] = sum;
            }
        }
    }
    printf("Matmul done: %lld\n", (long long)c[n * n - 1]);
    free(a); free(b); free(c);
    return 0;
}
